### PR TITLE
[chore] move logstransform processor to generated lifecycle tests

### PR DESCRIPTION
**Pull request by atoulme**
Related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27850

### CODE REVIEW

File: processor/logstransformprocessor/processor.go [hunk 1]
```diff
@@ -63,6 +63,9 @@ func (ltp *logsTransformProcessor) Capabilities() consumer.Capabilities {
 }
 
 func (ltp *logsTransformProcessor) Shutdown(_ context.Context) error {
+	if ltp.converter == nil {
```
Comment (TylerHelmuth) [inline processor/logstransformprocessor/processor.go:66]: I dont think we should be exiting early if `ltp.pipe` or `ltp.fromConverter` are not nil.

Comment (atoulme) [reply processor/logstransformprocessor/processor.go]: I have refactored the code a bit more to make the intent clear.

File: processor/logstransformprocessor/processor.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ type logsTransformProcessor struct {
 	converter     *adapter.Converter
 	fromConverter *adapter.FromPdataConverter
 	wg            sync.WaitGroup
+	cancelFn      func() error
```
Comment (bogdandrutu) [inline processor/logstransformprocessor/processor.go:36]: Maybe have a "shutdownFns []component.ShutdownFunc" and add to this slice after you create each of the components.

Comment (bogdandrutu) [reply processor/logstransformprocessor/processor.go:36]: If you don't like still at least rename this to stopFn of shutdownFn.

Comment (atoulme) [reply processor/logstransformprocessor/processor.go]: good stuff, done

**Review by djaglowski — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**